### PR TITLE
[IPAAS-287] Tests to verify advanced integration with multiple steps

### PR DESCRIPTION
**Pull request by lvydra**

### CODE REVIEW

File: e2e/app.po.ts [hunk 1]
```diff
@@ -103,7 +103,9 @@ export class AppPage {
 
   clickButton(buttonTitle: string): P<any> {
     log.info(`clicking button ${buttonTitle}`);
-    return this.getButton(buttonTitle).click();
+    const buttonElement = this.getButton(buttonTitle);
+    this.waitForElementToBePresent(buttonElement);
+    return buttonElement.click();
   }
```
Comment (dsimansk) [inline e2e/app.po.ts]: +1

File: e2e/app.po.ts [hunk 2]
```diff
@@ -195,6 +197,16 @@ export class AppPage {
...
+  waitForElementToBePresent(element) {
+    browser.wait(function () {
+        return element.isPresent();
+    }, 60000);
+
+    browser.wait(function () {
+        return element.isDisplayed();
+    }, 60000);
+  }
+
```
Comment (dsimansk) [inline e2e/app.po.ts]: I like the idea to wait for element in generatl. Maybe we could wrap this with `ExpectedConditions` and use either [`elementToBeClicable()`](https://github.com/angular/protractor/blob/master/lib/expectedConditions.ts#L188) or [`visibilityOf`](https://github.com/angular/protractor/blob/master/lib/expectedConditions.ts#L393). Wdyt?
Cc @jludvice

Comment (lvydra) [inline e2e/app.po.ts]: Sorry, I didn't notice we have functionality for that. 
What about that https://github.com/lvydra/syndesis-e2e-tests/commit/594ea73f0f897cb5dd0b56bf262e7990b3b8a9d1

File: e2e/app.po.ts [hunk 3]
```diff
@@ -103,7 +103,9 @@ export class AppPage {
 
   clickButton(buttonTitle: string): P<any> {
     log.info(`clicking button ${buttonTitle}`);
-    return this.getButton(buttonTitle).click();
+    const buttonElement = this.getButton(buttonTitle);
+    browser.wait(ExpectedConditions.presenceOf(buttonElement), 10000, 'Wait for button');
```
Comment (dsimansk) [inline e2e/app.po.ts]: Lets change this to `visibilityOf()` as it's a bit stronger. As it represents: "element is present in DOM && visible". Sorry for nit-picking here :)

Comment (jludvice) [inline e2e/app.po.ts]: I'm wondering what happens if `browser.wait` times out (in the context of https://github.com/syndesisio/syndesis-ui/issues/559).
We probably should return rejected promise with error data.
@lvydra can you check what happens if you wait for non-existent button? Is cucumber report generated or does it crash protractor and leave browser open ?

Comment (lvydra) [inline e2e/app.po.ts]: OK, I will check it.

Comment (lvydra) [inline e2e/app.po.ts]: I test it and it just crash and leave browser open.

**Comment by dsimansk** [comment]
@lvydra could you please try to implement the button wait in similar fashion. Seems to be working for me. Tests are finished with failure and report is generated. 
```
clickButton(buttonTitle: string): P<any> {
    log.info(`clicking button ${buttonTitle}`);
    return browser.wait(ExpectedConditions.visibilityOf(this.getButton(buttonTitle)), 6000, 'No button visible')
      .then(() =>  this.getButton(buttonTitle).click())
      .catch((e) => P.reject(e) );
  }
```

**Comment by lvydra** [comment]
OK

**Comment by lvydra** [comment]
Done plus "random" choosing from multiple links with same title added.

**Pull request merged**